scripts/build-platypus-csd3-ampere.sh

```diff
@@ -0,0 +1,270 @@
+
+}
+
+check_spack() {
+
+    cd "${ROOT_PATH}" || exit 1
+
+    if [ "$(command -v spack)" ]; then
+        echo "Spack command detected. Using pre-loaded spack."
+    elif [ -f "${ROOT_PATH}"/spack/share/spack/setup-env.sh ]; then
+        echo "Spack detected in root directory. Loading."
+        # shellcheck source=/dev/null
+        . spack/share/spack/setup-env.sh
+    else
+        echo "No spack detected. Building from source."
+        git clone --depth=100 https://github.com/spack/spack.git
+        # shellcheck source=/dev/null
+        . spack/share/spack/setup-env.sh
+    fi
+
+    check_cuda_version
+
+}
+
+install_spack_deps() {
+
+    # Cleaning up everything to start with a new environment
+    spack uninstall -ay arch=${ARCH}
+    spack clean -ab
+
+    echo "Installing Petsc..."
+    # Spack's petsc doesn't like openmpi, but it works with mpich
+    spack install petsc +cuda cuda_arch=80 +fortran +hdf5 +hypre +metis +mpi \
+        ^mpich +cuda cuda_arch=80 \
+        ^hdf5 +cxx +fortran +hl +mpi +shared \
+        ^hypre +mpi +shared +cuda cuda_arch=80 +superlu-dist +cublas +gpu-aware-mpi \
+        ^superlu-dist +cuda cuda_arch=80 +parmetis +shared
+    spack load petsc arch=${ARCH}
+
+    echo "Installing SLEPc..."
+    spack install slepc +cuda cuda_arch=80
+    spack load slepc arch=${ARCH}
+
+    echo "Installing netcdf..."
+    spack install netcdf-c +parallel-netcdf
+    spack load netcdf-c arch=${ARCH}
+
+    echo "Installing ninja..."
+    spack install ninja
+    spack load ninja arch=${ARCH}
+
+    echo "Adding python modules..."
+
+    spack install py-pyaml
+    spack load py-pyaml arch=${ARCH}
+
+    spack install py-jinja2
+    spack load py-jinja2 arch=${ARCH}
+
+    spack install py-packaging
+    spack load py-packaging arch=${ARCH}
+
+    spack install py-setuptools
+    spack load py-setuptools arch=${ARCH}
+
+}
+
+install_gslib() {
+
+    echo "Installing gslib..."
+    cd "${BUILD_PATH}" || exit 1
+    git clone https://github.com/Nek5000/gslib.git
+    cd gslib || exit 1
+    make CC=mpicc CFLAGS="-O2 -fPIC" -j"$compile_cores"
+}
+
+install_mfem() {
+
+    export CXX=mpic++
+    export CC=mpicc
+    export F90=mpif90
+    export F77=mpif77
+    export FC=mpif90
+
+    # Build MFEM
+    cd "${BUILD_PATH}" || exit 1
+    git clone https://github.com/mfem/mfem.git
+    cd mfem || exit 1
+    # This is just until MFEM merges Edward's changes. Without this, GPU build crashes!
+    git checkout EdwardPalmer99/add-missing-header-to-exodus-writer-fix
+    mkdir build
+    cd build || exit 1
+    echo "Building MFEM"
+    cmake .. \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DBUILD_SHARED_LIBS=YES \
+        -DMFEM_USE_OPENMP=NO \
+        -DMFEM_THREAD_SAFE=YES \
+        -DMFEM_ENABLE_EXAMPLES=YES \
+        -DMFEM_ENABLE_MINIAPPS=YES \
+        -DMFEM_USE_MPI=YES \
+        -DMFEM_USE_CUDA=YES \
+        -DCUDA_ARCH=sm_80 \
+        -DMFEM_USE_METIS_5=YES \
+        -DMFEM_USE_SUPERLU=YES \
```

Comment:
The new `build-platypus-csd3-ampere.sh` build script is currently failing for me on a new build when building MFEM:

> -- SuperLUDist: looking for required package: BLAS
-- Performing Test SuperLUDist_VERSION_OK
-- Performing Test SuperLUDist_VERSION_OK - Failed
-- Configuring incomplete, errors occurred!
MFEM build failed

Reply:
Resolved by addition of new SuperLU_Dist CMAKE flags